plugins/out_stackdriver/stackdriver_operation.c

```diff
@@ -64,81 +69,75 @@ int extract_operation(flb_sds_t *operation_id, flb_sds_t *operation_producer,
                       msgpack_object *obj, int *extra_subfields)
 {
     operation_status op_status = NO_OPERATION;
+    msgpack_object_kv *p;
+    msgpack_object_kv *pend;
+    msgpack_object_kv *tmp_p;
+    msgpack_object_kv *tmp_pend;
+
+    if (obj->via.map.size == 0) {    	
+        return FLB_FALSE;
+    }
+    p = obj->via.map.ptr;
+    pend = obj->via.map.ptr + obj->via.map.size;
+
+    for (; p < pend && op_status == NO_OPERATION; ++p) {
+
+        if (p->val.type != MSGPACK_OBJECT_MAP || p->key.type != MSGPACK_OBJECT_STR
```

Comment:
```suggestion
        if (p->val.type != MSGPACK_OBJECT_MAP 
```
This is checked by validate_key already.